### PR TITLE
[dask, sklearn] Fix predict proba.

### PR DESCRIPTION
**Pull request by trivialfis**
* For sklearn:
- Handles user defined objective function.
- Handles `softmax`.
* For dask:
- Use the implementation from sklearn, the previous implementation doesn't perform any extra handling.

### CODE REVIEW

**Comment by hcho3** [comment]
Please be more specific as to what is being fixed.

**Comment by trivialfis** [comment]
@hcho3 Added in PR description.  I need to add some more tests later.

**Comment by hcho3** [comment]
Thanks!

**Review by hcho3 — APPROVED** [review]
LGTM. Does this PR fix any of the open issues?

**Comment by trivialfis** [comment]
No, I found it myself.
